### PR TITLE
Update API specs n.a.v. RC1 bevindingen

### PR DESCRIPTION
**Pull request by sergei-maertens**
**ZRC**

* :pencil: `Zaak.einddatum` is `null` if it's unset
* :pencil: Notifications documentation added to API spec
* :boom: Normalized a bunch of enums that were left over 
  (only lowercase letters, numbers and underscores)
* :pencil: `Zaak.status` is `null` if it's unset
* :pencil: `Zaak.resultaat` is `null` if it's unset
* :boom: Added pagination to `KlantContact` list
* :boom: Added pagination to  `Resultaat` list
* :boom: Added pagination to `Rol` list
* :boom: Added pagination to `Status` list
* :boom: Added pagination to `ZaakObject` list
* :sparkles: Added `onderwerp` and `toelichting` to `KlantContact` resource
* :pencil2: Fixed a typo in the `Wijzigingen` schema (was `Wijzgingen`)

**DRC**

* :pencil: contains the requirements for webserver configuration so that file
  upload sizes are consistenly supported between providers
* :sparkles: `HTTP 413` as possible response header added (for uploads that exceed 
  the maximum body size)
* :pencil: Fixed documentation for notifications sent by DRC in API spec
* :boom: Removed audittrail from `ObjectInformatieObject`
* :boom: Removed notifications from `ObjectInformatieObject`
* :pencil2: Fixed a typo in the `Wijzigingen` schema (was `Wijzgingen`)

**ZTC**

* :pencil: Documented validation rules for `Zaaktype` create
* :boom: Normalized enums
* :pencil: Documented possible options for `ResultaatType.afleidingswijze` 
* :pencil: Documented possible options for `objecttype`

**BRC**

* :boom: Update operations on `BesluitInformatieObject` removed. Updates on each of 
  the attributes  were not allowed anyway, so the net result is the same.
* :pencil: Fixed response description HTTP 404
* :pencil2: Fixed typo in audit trail `Wijzigingen` resource

**NRC**

* :pencil: Fixed response description HTTP 404
* :pencil: Added missing documentation/descriptions

**AC**

* :pencil: Added missing documentation/descriptions
* :pencil: Fixed response description HTTP 404
* :pencil: Fixed polymorphic definition of component-specific authorizations

---

**Emoji legenda**

* :boom: Breaking change
* :bug: Bugfix
* :pencil2: Typo fixed
* :pencil: Documentation update
* :sparkles: New feature (addition, usually)

### CODE REVIEW

**Comment by michielverhoef** [comment]
Wat zijn de gevolgen van deze pullrequest? Is dit een reden om een RC2 uit te brengen of valt dit onder het rechtbreien van documentatie en toelichting?

**Comment by sergei-maertens** [comment]
@michielverhoef tot nu toe geen reden tot RC2 :-)

File: api-specificatie/drc/openapi.yaml [hunk 1]
```diff
@@ -26,6 +38,173 @@ info:
 security:
 - JWT-Claims: []
 paths:
+  /bestandsdelen/{uuid}:
```
Comment (sergei-maertens) [inline api-specificatie/drc/openapi.yaml]: This is part of the 1.1 release, MUST be removed again for 1.0 (self-reminder)

File: api-specificatie/drc/openapi.yaml [hunk 2]
```diff
@@ -3348,6 +3563,151 @@ components:
       scheme: bearer
       type: http
   schemas:
+    BestandsDeel:
```
Comment (sergei-maertens) [inline api-specificatie/drc/openapi.yaml]: Part of 1.1

**Comment by sergei-maertens** [comment]
@Hugo-ter-Doest beschrijving is bijgewerkt met alle changes voor RC2

File: api-specificatie/drc/openapi.yaml [hunk 3]
```diff
@@ -2,19 +2,31 @@ openapi: 3.0.0
...
+    \ kunnen aan andere objecten zoals zaken en besluiten worden\ngerelateerd (maar\
+    \ dat hoeft niet) en kunnen gebruiksrechten hebben.\n\nGEBRUIKSRECHTEN leggen\
     \ voorwaarden op aan het gebruik van het INFORMATIEOBJECT\n(buiten raadpleging).\
     \ Deze GEBRUIKSRECHTEN worden niet door de API gevalideerd\nof gehandhaafd.\n\n\
     De typering van INFORMATIEOBJECTen is in de Catalogi API (ZTC) ondergebracht in\n\
-    de vorm van INFORMATIEOBJECTTYPEn.\n\n**Afhankelijkheden**\n\nDeze API is afhankelijk\
-    \ van:\n\n* Catalogi API\n* Notificaties API\n* Autorisaties API *(optioneel)*\n\
+    de vorm van INFORMATIEOBJECTTYPEn.\n\n**Uploaden van bestanden**\n\nBinnen deze\
+    \ API bestaan een aantal endpoints die binaire data ontvangen, al\ndan niet base64-encoded.\
+    \ Webservers moeten op deze endpoints een minimale\nrequest body size van 4.0\
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply api-specificatie/drc/openapi.yaml]: Nee, maximum mogen ze zelf bepalen. Webservers moeten minimaal 4GB slikken, dan hebben alle clients garantie dat ze tot en met 4GB body size over de lijn kunnen gooien. Het voelt fout aan, maar het is juist :D

**Comment by sergei-maertens** [comment]
Ik merge en doe een stable release van alle componenten 👍

**Pull request merged**